src/Editor.ts

```diff
@@ -1,47 +1,89 @@
-import Destroyable from '@dojo/core/Destroyable';
 import global from '@dojo/core/global';
 import { createHandle } from '@dojo/core/lang';
+import { queueTask } from '@dojo/core/queue';
 import { debounce } from '@dojo/core/util';
+import { v } from '@dojo/widget-core/d';
+import { DNode, WidgetProperties } from '@dojo/widget-core/interfaces';
+import WidgetBase, { afterRender } from '@dojo/widget-core/WidgetBase';
+import { ThemeableMixin, ThemeableProperties, theme } from '@dojo/widget-core/mixins/Themeable';
 import project from './project';
+import * as css from './styles/editor.m.css';
 
 const globalMonaco: typeof monaco = global.monaco;
 
 /**
- * A class which is a simple abstraction of the `monaco-editor` editor
+ * @type EditorProperties
+ *
+ * Properties that can be set on Editor widget
+ *
+ * @property filename The filename (from the current `project`) that the editor should be displaying for editing
+ * @property options Editor options that should be passed to the monaco editor when it is created
+ * @property onEditorInit Called when the monaco editor is created and initialized by the widget, passing the instance of the monaco editor
+ * @property onEditorLayout Called when the widget calls `.layout()` on the monaco editor
  */
-export default class Editor extends Destroyable {
-	private _currentFile: string;
-	private _editor: monaco.editor.IStandaloneCodeEditor;
+export interface EditorProperties extends WidgetProperties, ThemeableProperties {
+	filename?: string;
+	options?: monaco.editor.IEditorOptions;
+
```

Comment:
well, they were methods, and not necessairly in order of the previous properties.  We have been putting them at the [end of properties](https://github.com/dojo/widgets/blob/master/src/checkbox/Checkbox.ts#L50) but I felt if I wan't going to order them alphabetically a new line would be a better visual indication of the "sectioning" of them.